backend/backend-schedule/src/main/java/eu/socialedge/hermes/backend/schedule/domain/gen/basic/BasicScheduleGenerator.java

```diff
@@ -115,21 +121,28 @@ private Trip generateTrip(int vehicleId, TimePoint timePoint) {
         val route = INBOUND.equals(timePoint.getDirection()) ? line.getInboundRoute() : line.getOutboundRoute();
         return new Trip(
             vehicleId,
-            calculateStops(timePoint.getTime(), route, averageSpeed, dwellTime));
+            calculateStops(timePoint.getTime(), route, averageSpeed));
     }
 
-    private List<Stop> calculateStops(LocalTime startTime, Route route, Quantity<Speed> averageSpeed, Duration dwellTime) {
+    private List<Stop> calculateStops(LocalTime startTime, Route route, Quantity<Speed> averageSpeed) {
         val stops = new ArrayList<Stop>();
 
         val averageSpeedValue = averageSpeed.to(Units.METRE_PER_SECOND).getValue().longValue();
 
-        stops.add(new Stop(startTime, startTime.plusSeconds(dwellTime.getSeconds()), route.iterator().next().getBegin()));
+        val headStation = route.iterator().next().getBegin();
```

Comment:
So look at the Route#getTail method, I think it returns first and getHead returns last station